src/MetaModels/Attribute/Color/Color.php

```diff
@@ -110,12 +111,17 @@ public function sortIds($idList, $strDirection)
      *
      * @return array
      */
-    private function colorSort($colors)
+    private function colorSort($colors, $strDirection)
     {
-        $counter = 0;
-        $sorted  = array();
+        $counter  = 0;
+        $sorted   = array();
+        $unsorted = array();
         foreach ($colors as $itemId => $colorValue) {
             $condensed = $colorValue[0];
+            if ($condensed == '') {
```

Comment:
Why do you keep track of the empty colors separately?
They will then not get sorted correctly.